지도 페이지 첫 커밋

nabitang12 opened the pull request:
## 📢 기능 설명

필요시 실행결과 스크린샷 첨부
<br>
![image](https://github.com/user-attachments/assets/36636755-5c03-4de3-a7d4-63d43c01c61c)

주요 기능들은  거의 구현한것 같습니다
스타일분리랑 헤더 씹히는 문제들이랑 사소한 기능들 구현하면 될 것 같습니다

## 연결된 issue

연결된 issue를 자동으로 닫기 위해 아래 {이슈넘버}를 입력해주세요. <br>
close #5 
<br>
<br>

## 🩷 Approve 하기 전 확인해주세요!

- [ ] 리뷰어가 확인해줬으면 하는 사항 적어주세요.
- [ ]

<br>

## ✅ 체크리스트

- [v ] PR 제목 규칙 잘 지켰는가?
- [ v] 추가/수정사항을 설명하였는가?
- [ v] 이슈넘버를 적었는가?
- [v ] Approve 하기 전 확인 사항 체크했는가?

nabitang12 committed:
지도 페이지 첫 커밋

Jayjunyoung commented on .env:
```diff
@@ -0,0 +1 @@
+VITE_API_KEY=AIzaSyBB1E1yqPNV9A2ig-g5tumXtzMy1m_vlE4
```
API키는 깃허브에 공개되면 안되서 .gitigore 파일에다가 해당 파일 명을 작성해주셔야 합니다!
(로컬에 .gitignore 라는 파일이 있어용)

Jayjunyoung commented on src/stores/mapStore.ts:
```diff
@@ -0,0 +1,25 @@
+  clearMarker: () => void
+}
+
+export const useStore = create<mapSlice>((set) => ({
```
스토어명 mapStore로 바꿔주세요!

Jayjunyoung commented on src/stores/searchStore.ts:
```diff
@@ -0,0 +1,11 @@
+import { create } from 'zustand'
+
+interface searchSlice {
```
타입스크립트에서 인터페이스명 같은 경우 대문자로 작성해주셔야하는데 어떤건 대문자 어떤건 소문자로 섞여 있는거 같애요! 대문자로 통일 해주시면 감사하겠습니다✨

Jayjunyoung replied:
시작할때 대문자로 시작하면 됩니다!
Ex) interface MovieState

Jayjunyoung commented on src/assets/mapIcon/말풍선_노랑.png:
이미지 파일명을 svg로 통일하기로 해서 바꿔줘야 할것 같습니다!
(저도 로컬상에서 통일해놓을려고 합니다)

Jayjunyoung commented on src/components/Debounce.tsx:
```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react'
+
+const Debounce = (value: string, delay: number) => {
```
```
함수명 export default function Debounce = () => {
 //해당 형식으로 바꿔주세요!!
}
```
Debounce적용 하는 컴포넌트 인가요 아니면 커스텀훅으로 작성해주신건가용?

nabitang12 replied:
리뷰 감사합니다! Debounce 커스텀훅입니다! 보내주신 리뷰 전부 고치고 다시 pr올리겠습니다!

Jayjunyoung replied:
커스텀훅이면 hooks 폴더에다 해주셔야 합니다! 그리고 커스텀훅 같은경우 앞에 명칭이 use로 시작해야되서 바꿔주시면 감사하겠습니다~

Jayjunyoung replied:
useSearchDebounce 이런식으로 바꿔주시면 좋을거 같네용

nabitang12 replied:
네 알겠습니다!

ssolfa commented:
수고하셨습니다!!! 혹시 pages 폴더 만들고 home 폴더도 만들어서 그 안에 `Map` 관련 페이지 넣어주실 수 있을까요??

<img width="1512" alt="스크린샷 2024-07-28 오후 10 13 16" src="https://github.com/user-attachments/assets/439e0989-bdfc-40be-acfe-db82b5e2d427">
그리고 시간 나시면 안쓰는 값들 삭제 부탁드립니다!

nabitang12 commented:
> 수고하셨습니다!!! 혹시 pages 폴더 만들고 home 폴더도 만들어서 그 안에 `Map` 관련 페이지 넣어주실 수 있을까요??
> 
> <img alt="스크린샷 2024-07-28 오후 10 13 16" width="1512" src="https://private-user-images.githubusercontent.com/87332713/352814561-439e0989-bdfc-40be-acfe-db82b5e2d427.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNzM3MzgsIm5iZiI6MTcyMjE3MzQzOCwicGF0aCI6Ii84NzMzMjcxMy8zNTI4MTQ1NjEtNDM5ZTA5ODktYmRmYy00MGJlLWFjZmUtZGI4MmI1ZTJkNDI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI4VDEzMzAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0ZWZjMWRiMTJlZmQ3ZDNjNzBhMjFmZTE4MjUyMTkzNWI5ZDNlNTJmNDJhZjFmZjZiMzcwZGM3YmU4YzMwZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7-y4Tf5LJkSw62HbsEeFjxQ3H_ZYu5GXp72nLqjNDjA"> 그리고 시간 나시면 안쓰는 값들 삭제 부탁드립니다!

nabitang12 commented:
![image](https://github.com/user-attachments/assets/195fd425-5db8-475a-938b-cd03091cb177)
이렇게 구분해놨었는데 제가 뭘 잘못한건지 아직 반영이 안된거 같습니다 ㅠ 안쓰는 값들도 삭제하겠습니다!

ssolfa commented:
>아직 반영이 안된거 같습니다 ㅠ 안쓰는 값들도 삭제하겠습니다!

커밋을 한 번 더 해주실 수 있나요? 그럼 바로 반영될 것 같습니다!

Jayjunyoung commented:
풀리퀘스트 이거 안 닫고 그냥 새로 커밋해서 푸쉬하면 고친 부분 반영됩니당 기윤님

nabitang12 committed:
Merge branch 'develop' of https://github.com/UMC-6th-BAB/BAB-FE into (#5)design/map-page

nabitang12 committed:
modified:   .eslintrc.json
	modified:   .gitignore
	renamed:    src/assets/greyIcon.ts -> src/assets/mapIcon/greyIcon.ts
	renamed:    src/assets/yellowIcon.ts -> src/assets/mapIcon/yellowIcon.ts
	renamed:    src/components/Header.tsx -> src/components/HeaderTitle/Header.tsx
	renamed:    src/components/AfterDiscountBar.tsx -> src/components/MapCard/DiscountCard/AfterDiscountBar.tsx
	renamed:    src/components/DiscountBar.tsx -> src/components/MapCard/DiscountCard/DiscountBar.tsx
	renamed:    src/components/DiscountFunction.tsx -> src/components/MapCard/DiscountCard/DiscountFunction.tsx
	renamed:    src/components/Map.tsx -> src/components/MapCard/GoogleMapCard/Map.tsx
	renamed:    src/components/BasicLayout.tsx -> src/components/MapCard/MapLayoutCard/MapLayout.tsx
	renamed:    src/components/AfterSearchBar.tsx -> src/components/MapCard/SearchCard/AfterSearchBar.tsx
	renamed:    src/components/SearchBar.tsx -> src/components/MapCard/SearchCard/SearchBar.tsx
	renamed:    src/components/Debounce.tsx -> src/hooks/Debounce.tsx
	new file:   src/pages/MapPage/MapRender.tsx
	modified:   src/stores/mapStore.ts
	modified:   src/stores/restaurentStore.ts
	modified:   src/stores/searchStore.ts

ssolfa commented on src/assets/mapIcon/greyIcon.ts:
이 부분 컴포넌트가 아닌 ts 파일로 만드신 이유가 궁금합니다!
또 styled-component가 아닌 css로 스타일 주신 이유도 궁금합니당....

nabitang12 replied:
컴포넌트로 해봤는데 아이콘에 내용이 입력이 안되가지고 저렇게 ts로 생성했습니다
css로 준 이유는 컴포넌트로 주면 또 입력이 안되가지고 css로 입력했습니다
 한번 더 다시 시도해 보겠습니다 ㅠ

ssolfa replied:
아니에요!!!! 저도 컴포넌트로 바꿔봤는데 안되는게 맞는 것 같습니다 ㅠ
이유 더 찾아보고 찾으면 공유드릴게요!!

ssolfa commented on src/components/MapCard/DiscountCard/DiscountBar.tsx:
```diff
@@ -0,0 +1,18 @@
+  font-size: 13px;
+  box-shadow: 0 0 3px;
+  z-index: 1;
+`
```
여기에 `text-align: center;` 추가하면 글씨 가운데 정렬 됩니다!

ssolfa commented on src/components/MapCard/DiscountCard/AfterDiscountBar.tsx:
```diff
@@ -0,0 +1,19 @@
+  font-size: 13px;
+  box-shadow: 0 0 3px;
+  z-index: 1;
+`
```
여기도 마찬가지로 `text-align: center;` 추가하면 글씨 가운데 정렬 됩니다!

ssolfa commented on src/components/MapCard/DiscountCard/DiscountFunction.tsx:
```diff
@@ -0,0 +1,27 @@
+import restaurantInfoStore from '@stores/restaurentStore'
+import { useStore } from '@stores/mapStore'
```
```suggestion
import { mapStore } from '@stores/mapStore'
```
이 부분 이렇게 import 하시고 싶으신게 맞나요??

ssolfa commented on src/stores/mapStore.ts:
```diff
@@ -0,0 +1,25 @@
+  lat: number
+  lng: number
+  markers: google.maps.marker.AdvancedMarkerElement[]
+  googleMap: google.maps.Map
```
```suggestion
  googleMap: google.maps.Map | null
```
googleMap을 초기화 시에는 null로 설정해서 오류 수정할 수 있을 것 같습니다!

ssolfa commented on src/stores/mapStore.ts:
```diff
@@ -0,0 +1,25 @@
+  markers: [],
+  lat: 37.496336,
+  lng: 126.95733,
+  addMarker: (marker) =>
```
이부분에도 ` googleMap: null,` 추가해서 초기값 설정해주면 오류 수정할 수 있을 것 같습니다!

Jayjunyoung commented on src/hooks/Debounce.tsx:
여기 useDebounceSearch로 바꿔주세요!
파일명이랑 함수명

nabitang12 committed:
modified:   src/assets/mapIcon/greyIcon.ts
	modified:   src/assets/mapIcon/yellowIcon.ts
	modified:   src/components/MapCard/DiscountCard/AfterDiscountBar.tsx
	modified:   src/components/MapCard/DiscountCard/DiscountBar.tsx
	modified:   src/components/MapCard/DiscountCard/DiscountFunction.tsx
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/components/MapCard/SearchCard/AfterSearchBar.tsx
	modified:   src/components/MapCard/SearchCard/SearchBar.tsx
	modified:   src/components/StatusBar/StatusBar.style.ts
	renamed:    src/hooks/Debounce.tsx -> src/hooks/Debounce.ts
	modified:   src/stores/mapStore.ts
	modified:   src/stores/restaurentStore.ts

yongmin01 commented on src/components/StatusBar/StatusBar.style.ts:
```diff
@@ -9,9 +9,9 @@ export const StatusBarContainer = styled.div`
+  position: fixed;
   top: 0;
-  left: 0;
+  left: 25px;
```
이부분은 왜 변경하신건가용?? 적용했을 때 문제 없던데...!

nabitang12 replied:
저번에 회의때 header를 position:fixed 로 고친다고 하셔서 고쳤는데 잘 안됐던것 같습니다 다시 해보겠습니다!

yongmin01 commented on src/components/HeaderTitle/Header.tsx:
안 쓰시는 컴포넌트면 삭제하는게 좋을 것 같아요! Header.style 파일두요!!

nabitang12 committed:
deleted:    src/components/HeaderTitle/Header.tsx
	modified:   src/components/MapCard/DiscountCard/AfterDiscountBar.tsx
	new file:   src/components/MapCard/DiscountCard/DiscountBar.style.ts
	modified:   src/components/MapCard/DiscountCard/DiscountBar.tsx
	deleted:    src/components/MapCard/DiscountCard/DiscountFunction.tsx
	new file:   src/components/MapCard/GoogleMapCard/Map.style.ts
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	deleted:    src/components/MapCard/MapLayoutCard/MapLayout.tsx
	modified:   src/components/MapCard/SearchCard/AfterSearchBar.tsx
	new file:   src/components/MapCard/SearchCard/SearchBar.style.ts
	modified:   src/components/MapCard/SearchCard/SearchBar.tsx
	modified:   src/components/StatusBar/StatusBar.style.ts

nabitang12 commented:
오늘 올린커밋은 제가 몸이 아파서 대충했습니다 내일 다시 정정해서 커밋 올리도록 하겠습니다 죄송합니다

nabitang12 committed:
filter 함수 오류 수정 css 분리 작업

nabitang12 commented:
filter 함수 오류 수정 , css 분리 작업 완료했습니다 마커 클러스터링은 구현이 힘들것 같아서 차라리 줌인 상태에서 아이콘을 크게 보여주고 줌아웃 상태에서 아이콘을 작게 하는걸로 생각해보겠습니다

nabitang12 committed:
modified:   src/assets/mapIcon/greyIcon.ts
	modified:   src/pages/MapPage/MapRender.tsx

nabitang12 committed:
Merge branch 'develop' into (#5)design/map-page

nabitang12 committed:
new file:   src/assets/mapIcon/smallGreyIcon.ts
	new file:   src/assets/mapIcon/smallYellowIcon.ts
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/pages/MapPage/MapRender.tsx

nabitang12 committed:
Merge branch '(#5)design/map-page' of https://github.com/UMC-6th-BAB/BAB-FE into (#5)design/map-page

nabitang12 committed:
modified:   package-lock.json
	modified:   src/App.tsx
	modified:   src/components/AppContent.tsx
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/pages/MapPage/MapRender.tsx

nabitang12 committed:
Merge branch 'develop' into (#5)design/map-page

nabitang12 committed:
modified:   src/components/AppContent.tsx

nabitang12 committed:
Merge branch '(#5)design/map-page' of https://github.com/UMC-6th-BAB/BAB-FE into (#5)design/map-page
